packages/shared_preferences/shared_preferences_foundation/lib/shared_preferences_foundation.dart

```diff
@@ -2,106 +2,351 @@
+    try {
+      return await _api.getValue(key, pigeonOptions) as double?;
+    } on PlatformException catch (err) {
+      if (err.code == _argumentErrorCode) {
+        throw ArgumentError(
+            'shared_preferences_foundation getString argument error${err.message ?? ''}');
+      } else {
+        rethrow;
+      }
+    }
+  }
+
+  @override
+  Future<int?> getInt(
+    String key,
+    SharedPreferencesOptions options,
+  ) async {
+    final SharedPreferencesPigeonOptions pigeonOptions =
+        _convertOptionsToPigeonOptions(options);
+
+    try {
+      return await _api.getValue(key, pigeonOptions) as int?;
+    } on PlatformException catch (err) {
+      if (err.code == _argumentErrorCode) {
+        throw ArgumentError(
+            'shared_preferences_foundation getString argument error${err.message ?? ''}');
+      } else {
+        rethrow;
+      }
+    }
+  }
+
+  @override
+  Future<List<String?>?> getStringList(
+    String key,
+    SharedPreferencesOptions options,
+  ) async {
+    final SharedPreferencesPigeonOptions pigeonOptions =
+        _convertOptionsToPigeonOptions(options);
+
+    try {
+      return _api.getStringList(key, pigeonOptions);
+    } on PlatformException catch (err) {
+      if (err.code == _argumentErrorCode) {
+        throw ArgumentError(
+            'shared_preferences_foundation getString argument error${err.message ?? ''}');
+      } else {
+        rethrow;
+      }
+    }
+  }
+
+  @override
+  Future<bool> clear(
+    ClearPreferencesParameters parameters,
+    SharedPreferencesOptions options,
+  ) async {
+    final PreferencesFilters filter = parameters.filter;
+    try {
+      await _api.clear(
+        filter.allowList?.toList(),
+        _convertOptionsToPigeonOptions(options),
+      );
+    } on PlatformException catch (err) {
+      if (err.code == _argumentErrorCode) {
+        throw ArgumentError(
+            'shared_preferences_foundation getString argument error${err.message ?? ''}');
+      } else {
+        rethrow;
+      }
+    }
     return true;
   }
+
+  @override
+  Future<Map<String, Object>> getPreferences(
+    GetPreferencesParameters parameters,
+    SharedPreferencesOptions options,
+  ) async {
+    final PreferencesFilters filter = parameters.filter;
+    try {
+      final Map<String?, Object?> data = await _api.getAll(
+        filter.allowList?.toList(),
+        _convertOptionsToPigeonOptions(options),
+      );
+      return data.cast<String, Object>();
+    } on PlatformException catch (err) {
+      if (err.code == _argumentErrorCode) {
+        throw ArgumentError(
+            'shared_preferences_foundation getString argument error${err.message ?? ''}');
+      } else {
+        rethrow;
+      }
+    }
+  }
+}
+
+/// Options for the Foundation specific SharedPreferences plugin.
+class SharedPreferencesFoundationOptions extends SharedPreferencesOptions {
+  /// Creates a new instance with the given options.
+  SharedPreferencesFoundationOptions({
```

Comment:
`const`, and `@immutable` for the class?

Reply:
const is already immutable isn't it?

Reply:
can't be const and have throws in the constructor anyway